dag_in_context/src/utility/util.egg

```diff
@@ -55,3 +55,67 @@
       ((union (Get (Concat expr1 expr2) i)
               (Get expr2 (- i len1))))
        :ruleset always-run)
+
+
+;; This Expr size actually do the same thing as cost function.
+(function Expr-size (Expr) i64 :unextractable :merge (min old new) )
```

Comment:
Maybe leave a note here that this computes the tree size (we don't need to say this before but now we are using a DAG semantics)